docs/components/codeblock.md

```diff
@@ -0,0 +1,268 @@
+# CodeBlock
+
+**KCodeBlock** is a code block component with built-in search and filter functionalities.
+
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+/>
+```
+
+## Props
+
+### id
+
+* **Type**: `string`
+* **Required**: no
+* **Default**: `'code-block'`
+
+An ID used to identify form elements like the search input and its label.
+
+### code
+
+* **Type**: `string`
+* **Required**: yes
+
+The code that will be rendered as a text node inside of a `<pre><code></code></pre>` fragment.
+
+### language
+
+* **Type**: `string`
+* **Required**: yes
+
+The syntax language of `props.code` (e.g. `'json'`).
+
+### isSearchable
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Shows an actions bar with a search input and related action buttons.
+
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+```
+
+### isProcessing
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Allows controlling the processing state from outside the component. This allows a parent component to show the processing icon when it’s, for example, currently syntax highlighting the code.
+
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-searchable
+  is-processing
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-processing
+/>
+```
+
+### query
+
+* **Type**: `string`
+* **Required**: no
+* **Default**: `''`
+
+Used as the initial value of the search input. Can be used to initialize a code block with a query which was read from client storage.
+
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="key"
```

Comment:
suggestion: let's change the initial query to a term that is present, e.g. `vite`

Reply:
Done. That was an artifact of changing the example code. I might change it again to be a bit more generic.